homeassistant/components/nibe_heatpump/config_flow.py

```diff
@@ -0,0 +1,128 @@
+"""Config flow for Nibe Heat Pump integration."""
+from __future__ import annotations
+
+import errno
+from typing import Any
+
+from nibe.connection.nibegw import NibeGW
+from nibe.exceptions import CoilReadException, CoilWriteException
+from nibe.heatpump import HeatPump, Model
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_IP_ADDRESS, CONF_MODEL
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import config_validation as cv
+from homeassistant.util.network import is_ipv4_address
+
+from .const import (
+    CONF_CONNECTION_TYPE,
+    CONF_CONNECTION_TYPE_NIBEGW,
+    CONF_LISTENING_PORT,
+    CONF_REMOTE_READ_PORT,
+    CONF_REMOTE_WRITE_PORT,
+    CONF_WORD_SWAP,
+    DOMAIN,
+    LOGGER,
+)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_MODEL): vol.In([e.name for e in Model]),
+        vol.Required(CONF_IP_ADDRESS): str,
```

Comment:
you can put a function that matches a regex here as a validator

Reply:
In normal validators yes, for config flows its much more limited.